### DISCUSSION TITLE
Lines are breaking after textwidth

### DISCUSSION THREAD

**Discussion by MajorDallas** [neovim/neovim#30290] [Q&A] [answered]
With `tw=80 fo=tcq] fp= fex=` and a line where the last word starts on a column less than `tw`, nvim will only break the line if the last column in the word is above 85 or so. This applies to insert mode and the `gq` command. 

I think it's also new behavior: it used to reliably break on the whitespace _before_ that word (while typing, as soon as it hit column 81); now, it breaks on the space _after_ the word or at column 85, whichever comes first. I noticed the change after setting up on a new workstation back in June and thought I'd try it for a while. I don't remember what version of nvim I was running on my old workstation--I want to say it was 0.9.something? The current version is v0.11.0-dev.

Anyway, I have finally decided I don't like the new behavior. Now I'm trying to revert to the old behavior and can't seem to find any combination of format options to do it (with/without `]` or `l` or `1`, etc., but I have not tried `a` or `w`). The only thing I've found so far that works is to set `tw=77` or lower. 

Is there some new "give or take columns = 5" option that was added in the last couple releases, and how do I set it back to 0?

**Comment by zeertzjq** [comment]
There is no such behavior change from Nvim, so it should come from something else.

**Reply by MajorDallas** [reply, marked as answer]
Agh, you're right, nvim was lying to me because of `linebreak`. Apparently the "virtual character" (idk what else to call it) that is inserted counts for potentially several columns, so going to the end of the line might show 85 in the status line even though the line is really 79 columns. 

With `nolinebreak`:
```
        msg (str): The string that will become the 'result' field of the audi
t
# after $, the col number is 78
```
and with `linebreak`:

```
        msg (str): The string that will become the 'result' field of the 
audit
# after $, the col number is 82
```

This has only been a nuisance in the last couple of months, which is why I thought it was some changed behavior... I may have to take it up with vim-airline to see if they changed the default statusline format to use the "virtual column."
